### PR TITLE
Enable DM to the Slack Bot

### PR DESCRIPTION
**Pull request by lasryaric**

### CODE REVIEW

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 1]
```diff
@@ -24,12 +24,14 @@ type SlackWebhookReqBody = {
...
     channel?: string;
+    subtype: "message_changed";
     user?: string;
     ts?: string; // slack message id
     thread_ts?: string; // slack thread id
     type?: string; // event type (eg: message)
-    channel_type?: string; // channel type (eg: channel, im, mpim)
+    channel_type?: "channel" | "im" | "mpim";
     text: string; // content of the message
+    bot_profile: unknown;
```
Comment (philipperolet) [inline connectors/src/api/webhooks/webhook_slack.ts:34]: to remove

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 2]
```diff
@@ -24,12 +24,14 @@ type SlackWebhookReqBody = {
   event?: {
     bot_id?: string;
     channel?: string;
+    subtype: "message_changed";
```
Comment (philipperolet) [inline connectors/src/api/webhooks/webhook_slack.ts:27]: subtype? (q.mark to add)

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { Request, Response } from "express";
 import { botAnswerMessageWithErrorHandling } from "@connectors/connectors/slack/bot";
 import {
   getAccessToken,
-  whoAmI,
+  whoAmIMemoized,
```
Comment (philipperolet) [inline connectors/src/api/webhooks/webhook_slack.ts:6]: getBotUserId

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 4]
```diff
@@ -41,6 +43,53 @@ type SlackWebhookResBody =
   | null
   | APIErrorWithStatusCode;
 
+async function handleChatBot(req: Request, res: Response) {
+  const slackMessage = req.body.event.text;
```
Comment (philipperolet) [inline connectors/src/api/webhooks/webhook_slack.ts:47]: optional: you could do a bit more concise this way
const slackRequest = { message: , teamId: , ..., messageTs: req.body.event.ts}

if (Object.values(slackRequest).some(v => !v)) { // error "missing fileds"

const botRes = await botAnswer..(slackRequest)
logger.error({error: ..., slackRequest)

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 5]
```diff
@@ -141,82 +147,123 @@ const _webhookSlackAPIHandler = async (
...
+                message: `Connector ${slackConfig.connectorId} not found`,
+              },
+              status_code: 404,
+            });
+          }
+          const slackAccessToken = await getAccessToken(connector.connectionId);
+          const myUserId = await whoAmIMemoized(slackAccessToken);
+          if (req.body.event?.user === myUserId) {
+            // Message sent from the bot itself.
+            return res.status(200).send();
```
Comment (philipperolet) [inline connectors/src/api/webhooks/webhook_slack.ts:185]: curiosity: I thought the message sent by the bot was dumped via the subtype: "message_created"; why do we need both?

**Review by philipperolet — CHANGES_REQUESTED** [review]
:+1: LGTM appart from the changes we discussed; left an optional refactoring, and a curiosity question

**Comment by lasryaric** [comment]
@philipperolet merging because I want to deploy quickly as I have to leave early and need to test internally + deploy to our first design partner.

Please comment if there is additional change, I will take care of them. 

thanks.

**Pull request merged**